src/System.Drawing.Common/tests/Printing/PaperSizeTest.cs

```diff
@@ -0,0 +1,167 @@
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// Author:
+// 	Andy Hume <andyhume32@yahoo.co.uk>
+//
+
+using Xunit;
+
+namespace System.Drawing.Printing.Test
+{
+    public class PaperSizeTest
+    {
+
+        [Fact]
+        public void ChangeHeight_OnCustomPaperKind_ShouldNotThrowAnExcetion()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.Height = 1;
+
+            Assert.Equal(1, ps.Height);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnFixedPaperKind_ShouldThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = (int)PaperKind.A4;
+
+            Assert.Throws<ArgumentException>(() => ps.Height = 2);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnNotRealCustomKind_ShouldThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = 999999;
+
+            // The properties can be changed only when the *real* Kind is Custom 
+            // and not when is 'effectively' Custom.
+            Assert.Throws<ArgumentException>(() => ps.Height = 4);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnCustomKind_ShouldNotThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = (int)PaperKind.Custom;
+            ps.Height = 1;
+
+            Assert.Equal(1, ps.Height);
+        }
+
+        [Fact]
+        public void ChangeHeight_OnComputedCustomKind_ShouldThrowAnException()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 100);
+
+            ps.RawKind = 1 + (int)PaperKind.PrcEnvelopeNumber10Rotated;
+
+            Assert.Equal(PaperKind.Custom, ps.Kind);
+            Assert.Throws<ArgumentException>(() => ps.Height = 9);
+        }
+
+        [Fact]
+        public void Ctor()
+        {
+            PaperSize ps = new PaperSize();
+
+            Assert.Equal(PaperKind.Custom, ps.Kind);
+            Assert.Equal(string.Empty, ps.PaperName);
+        }
+
+        [Fact]
+        public void Param_Ctor()
+        {
+            PaperSize ps = new PaperSize("foo", 100, 200);
+
+            Assert.Equal(PaperKind.Custom, ps.Kind);
+            Assert.Equal(200, ps.Height);
+            Assert.Equal(100, ps.Width);
+        }
+
+        [Fact]
+        public void PaperSizeRawKindTest()
```

Comment:
I think this could be converted into a `Theory` method, because it seems to be testing a bunch of things in sequence, which aren't necessarily related to each other.